stdlib/ballerina-jms/src/main/java/org/ballerinalang/net/jms/JMSUtils.java

```diff
@@ -99,10 +99,25 @@ public static Connection createConnection(Struct connectionConfig) {
         Properties properties = new Properties();
         configParams.forEach(properties::put);
 
+        //check for additional jndi properties
+        Map<String, Value> props = connectionConfig.getMapField(Constants.PROPERTIES_MAP);
+        if (props != null) {
+            for (Map.Entry<String, Value> entry : props.entrySet()) {
+                properties.put(entry.getKey(), entry.getValue().getStringValue());
+            }
+        }
+
         try {
             InitialContext initialContext = new InitialContext(properties);
             ConnectionFactory connectionFactory = (ConnectionFactory) initialContext.lookup(factoryName);
-            return connectionFactory.createConnection();
+            String username = connectionConfig.getStringField(Constants.ALIAS_USERNAME);
+            String password = connectionConfig.getStringField(Constants.ALIAS_PASSWORD);
+
+            if (!username.isEmpty() && !password.isEmpty()) {
```

Comment:
As we have specified username and passwords are nullable fields in connection.bal [1]. Shall we also include the null check as well. Unless it'll lead to a NPE when the user doesn't provide a username or password. You can use method[2] perform this operation.

[1]. https://github.com/ballerina-platform/ballerina-lang/pull/9212/files#diff-1c8397e65ef4ec76bbff9c22853bdf0cR57
[2]. https://github.com/ballerina-platform/ballerina-lang/blob/master/stdlib/ballerina-jms/src/main/java/org/ballerinalang/net/jms/JMSUtils.java#L147